ors-engine/src/main/java/org/heigit/ors/matrix/MatrixRequest.java

```diff
@@ -224,9 +222,9 @@ else if (getSearchParameters().getDynamicSpeeds() && routingProfile.getGraphhopp
         } catch (PointNotFoundException e) {
             throw e;
         } catch (MaxVisitedNodesExceededException e) {
-            throw new InternalServerException(MatrixErrorCodes.MAX_VISITED_NODES_EXCEEDED, "Unable to compute a distance/duration matrix: " + e.getMessage());
+            throw new StatusCodeException(400, MatrixErrorCodes.MAX_VISITED_NODES_EXCEEDED, "Unable to compute a distance/duration matrix: " + e.getMessage());
```

Comment:
According to [Wikipedia](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) 4xx status codes indicate client errors. However, exceeding of the maximum allowed visited nodes is a server-side error. As such, it should return a 5xx error.

Reply:
fixed.